src/tools/tools.R

```diff
@@ -97,6 +97,16 @@ get.print.str <- function(data) {
   return(paste0(capture.output(print(data)), collapse = "\n"))
 }
 
+#' Just like `get.print.str()` but providing a 5-chars indentation for logging.
+#'
+#' @param data Object to be printed.
+#'
+#' @returns A single string, with indentation chars.
+#'
+#' @author FeAR - MrHedmad
+get.print.ind <- function(data) {
+  return(paste0(capture.output(print(data)), collapse = "\n  || "))
+}
```

Comment:
I feel that the `get.print.ind` addition is not that useful as the data snippets are *not* logged to the logfile (they would not be useful in any case) but are just presented to the user, most often during a slowmode run.
Additionally, snippets of data do not conform to the "one line per log message" we were going for. If we *do* want snippets of data to be logged for reference, opening a new logfile (perhaps with the same name and path as the original, but with an additional bit? `original_name.data.log`?) and parsing them there could be ok.

To do this, we would need a suite of functions. I'd do them along these lines:
```r
get_datetime_str <- function() {return(paste("[", Sys.time(), Sys.date(), "]"))}

# This makes a function that appends strings to the specified log file
# With the given headers, footers and padding (applied at the start of every
# line pushed to the data log)
data_log_maker <- function(
  log_path, header = get_datetime_str(), footer = "-------", padding = "    "
) {
  ... # I'm not going to implement it now...
}

# It would be called like such:
push_to_datalog <- data_log_maker("/path/to/some_log.data.log")
push_to_datalog(get.print.str(some_data))
```

Reply:
Totally agree. Let's go for "one line per log message"... but the idea of a data.log sounds good to me. I think that some of that snippets are worth being saved. Moreover, the current implementation of 'GATTACA run' _does_ write out design matrix, filtering report, and other snippets, in multiple lines, both on screen and log file. So we could think to harmonize this behavior.

Reply:
It does?? It shouldn't... I'll look into it, and implement the data log while I'm at it.